sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/firestore/FirestoreV1FnRunQueryTest.java

```diff
@@ -283,62 +292,86 @@ protected RunQueryFn getFn(
-      FieldReference foo = FieldReference.newBuilder().setFieldPath(fieldPath).build();
+    public TestData(String projectId, Function<FieldReference, List<Order>> orderFunction,
+        Filter filter) {
       StructuredQuery.Builder builder =
           StructuredQuery.newBuilder()
               .addFrom(
                   CollectionSelector.newBuilder()
                       .setAllDescendants(false)
                       .setCollectionId("collection"))
-              .setWhere(
-                  Filter.newBuilder()
-                      .setFieldFilter(
-                          FieldFilter.newBuilder()
-                              .setField(foo)
-                              .setOp(Operator.EQUAL)
-                              .setValue(Value.newBuilder().setStringValue("bar"))
-                              .build()));
-
-      orderFunction.apply(foo).forEach(builder::addOrderBy);
+              .setWhere(filter);
+
+      orderFunction.apply(FILTER_FIELD_PATH).forEach(builder::addOrderBy);
       request =
           RunQueryRequest.newBuilder()
               .setParent(String.format("projects/%s/databases/(default)/document", projectId))
               .setStructuredQuery(builder)
               .build();
 
-      response1 = newResponse(fieldPath, 1);
-      response2 = newResponse(fieldPath, 2);
-      response3 = newResponse(fieldPath, 3);
+      response1 = newResponse(1);
+      response2 = newResponse(2);
+      response3 = newResponse(3);
     }
 
-    private static RunQueryResponse newResponse(String field, int docNumber) {
+    /**
+     * Returns single-document response like this:
+     * {
+     *   "__name__": "doc-{docNumber}"
+     *   "foo": "bar"
+     *   "foobar": {"bar": "foo"}
+     * }
+     */
+    private static RunQueryResponse newResponse(int docNumber) {
       String docId = String.format("doc-%d", docNumber);
       return RunQueryResponse.newBuilder()
           .setDocument(
               Document.newBuilder()
                   .setName(docId)
                   .putAllFields(
-                      ImmutableMap.of(field, Value.newBuilder().setStringValue("bar").build()))
-                  .build())
+                      ImmutableMap.of(
+                          "foo", Value.newBuilder().setStringValue("bar").build(),
+                          "foobar", Value.newBuilder().setMapValue(MapValue.newBuilder()
+                              .putFields("bar", Value.newBuilder().setStringValue("foo").build())
+                              .build()).build()
+                      )))
           .build();
     }
 
     private static Builder fieldEqualsBar() {
-      return new Builder();
+      return new Builder().setFilter(Filter.newBuilder()
+          .setFieldFilter(
+              FieldFilter.newBuilder()
+                  .setField(FILTER_FIELD_PATH)
+                  .setOp(Operator.EQUAL)
+                  .setValue(Value.newBuilder().setStringValue("bar"))
```

Comment:
maybe just declare two Filter constants FOO_EQUALS_BAR and FOO_NOT_EQUALS_FOO around line 297? there's no argument to these two methods so everytime these methods get called they just return the same filter, constants will do just as good as methods.

Reply:
Sure why not. Done.